File: 520.检测大写字母.java
/*
 * @lc app=leetcode.cn id=520 lang=java
 *
 * [520] 检测大写字母
 */

// @lc code=start
class Solution {
    public boolean detectCapitalUse(String word) {
        int num = 0;
        for (int i = 0; i< word.length(); i++) {
            if (Character.isUpperCase(word.charAt(i))){
                num++;
            }
        }
        return num == 0 || num == word.length() 
        || num == 1 && Character.isUpperCase(word.charAt(0));
    }
}
// @lc code=end

File: 520.检测大写字母.py
#
# @lc app=leetcode.cn id=520 lang=python3
#
# [520] 检测大写字母
#

# @lc code=start


class Solution:
    def detectCapitalUse(self, word: str) -> bool:
        daxie = 0
        for ch in word:
            if ch.isupper():
                daxie += 1
        return daxie == 0 or daxie == len(word) or daxie == 1 and word[0].isupper()

# @lc code=end
